src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -287,6 +265,29 @@ void GinIndexStore::cancel() noexcept
         postings_file_stream->cancel();
 }
 
+void GinIndexStore::initSegmentId()
+{
+    String segment_id_file_name = getName() + GIN_SEGMENT_ID_FILE_TYPE;
+
+    UInt32 result;
+    if (storage->existsFile(segment_id_file_name))
```

Comment:
L. 275-284 and l. 206-215 are the same as far as I see, can we factorize them into a helper function?